libs/utils/src/http/endpoint.rs

```diff
@@ -32,6 +32,36 @@ static X_REQUEST_ID_HEADER: HeaderName = HeaderName::from_static(X_REQUEST_ID_HE
 #[derive(Debug, Default, Clone)]
 struct RequestId(String);
 
+/// Adds a tracing info_span! instrumentation around the handler events.
+/// Use this to distinguish between logs of different HTTP requests.
```

Comment:
Could add a comment that this is the only way we can attach a span for the duration of the request handling with this framework vs. enter/exit points accessible with middleware.

```suggestion
/// Adds a tracing info_span! instrumentation around the handler events.
///
/// Use this to distinguish between logs of different HTTP requests.
///
/// Instrumenting the request handling future with a span is the only way
/// with $current_framework because middlewares only have access to
/// pre/post points of a request handling lifecycle, not to the actual
/// future which needs to be instrumented to use a span.
```

Perhaps it would work as a comment in `pub async fn handle` below just as well.

Reply:
I've added some paraphrased version of the PR message, but I expect it to be added as a part of the final squash commit.

Reply:
Yeah that [c5908f9](https://github.com/neondatabase/neon/pull/3755/commits/c5908f942b5b535a35982a9ee2febb886066de94) is good for me.